test/Dialect/Calyx/emit.mlir

```diff
@@ -241,3 +241,40 @@ module attributes {calyx.entrypoint = "main"} {
     }
   }
 }
+
+// -----
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%clk: i1 {clk}, %reset: i1 {reset}, %go: i1 {go}) -> (%out0: i32, %out1: f32, %done: i1 {done}) {
+    %c42_i32 = hw.constant 42 : i32
+    %cst = calyx.constant 4.200000e+00 : f32
+    %true = hw.constant true
+    %ret_arg1_reg.in, %ret_arg1_reg.write_en, %ret_arg1_reg.clk, %ret_arg1_reg.reset, %ret_arg1_reg.out, %ret_arg1_reg.done = calyx.register @ret_arg1_reg : f32, i1, i1, i1, f32, i1
+    %ret_arg0_reg.in, %ret_arg0_reg.write_en, %ret_arg0_reg.clk, %ret_arg0_reg.reset, %ret_arg0_reg.out, %ret_arg0_reg.done = calyx.register @ret_arg0_reg : i32, i1, i1, i1, i32, i1
+    calyx.wires {
+      calyx.assign %out1 = %ret_arg1_reg.out : f32
+      calyx.assign %out0 = %ret_arg0_reg.out : i32
+
+      // CHECK-LABEL: group ret_assign_0 {
+      // CHECK-NEXT:   ret_arg0_reg.in = 32'd42;
+      // CHECK-NEXT:   ret_arg0_reg.write_en = 1'd1;
+      // CHECK-NEXT:   ret_arg1_reg.in = 32'd4.200000;
```

Comment:
Wait this well formed in Calyx?

Reply:
You are right that it's not well-formed currently. I've been using json data to read external memory data, which can be floating points.
I will support direct floating point representation in Calyx in the future.

Reply:
I think floating-point constants are tricky and we should be careful about adding them to the language. A particular challenge is that we'd need to have an IEEE-754 (floating point specification) compatible emitter which can generate the right bit patterns for a given floating-point number which can be easy to get wrong. I've created a new issue in the Calyx repo to discuss this: https://github.com/calyxir/calyx/issues/2061

In the meantime, one approach I recommend is adding the ability to generate the right bit patterns for the floating-point number and generating that in the emitter to native Calyx. Does that make sense?

Reply:
> In the meantime, one approach I recommend is adding the ability to generate the right bit patterns for the floating-point number and generating that in the emitter to native Calyx

Do you mean something like [this](https://github.com/llvm/circt/blob/437693a508e974a43982a1a39d887feff7107342/test/Dialect/Calyx/emit.mlir#L261)?